### PR TITLE
[INFINITY-588] Restart task endpoint

### PR DESCRIPTION
**Pull request by mgummelt**
Moved /v1/state/tasks to /v1/tasks
Added POST /v1/tasks/restart/<task-name>?replace=<replace>

### CODE REVIEW

**Comment by mgummelt** [comment]
cc @gabrielhartmann

File: src/main/java/org/apache/mesos/scheduler/api/TaskResource.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    private static final Logger logger = LoggerFactory.getLogger(TaskResource.class);
+
+    private final StateStore stateStore;
+    private final PropertyDeserializer propertyDeserializer;
+    private final TaskKiller taskKiller;
+    private final String frameworkName;
+
+    /**
+     * Creates a new TaskResource which cannot deserialize Properties. Callers will receive a
+     * "204 NO_CONTENT" HTTP response when attempting to view the content of a property.
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/scheduler/api/TaskResource.java]: Looks like all the property handling is still over in the `StateResource`.  Was the intent to replace the `StateResource` with the `TaskResource`?

File: src/main/java/org/apache/mesos/state/api/StateResource.java [hunk 2]
```diff
@@ -75,122 +69,6 @@ public Response getFrameworkId() {
...
-            }
-        }
-        return String.join(",", ports);
-    }
-
-
-    /**
-     * Produces a listing of the names of all stored tasks.
-     */
     @Path("/properties")
```
Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/state/api/StateResource.java]: The `TaskResource` constructors make reference to properties, but don't appear to actually serve property query responses.  Needs to be resolved one way or the other.  Was the intention of the `StateResource` to now solely present properties?

File: src/main/java/org/apache/mesos/scheduler/api/TaskResource.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        this(stateStore, null, taskKiller, frameworkName);
+    }
+
+    /**
+     * Creates a new TaskResource which can deserialize Properties. Callers will be able to view
+     * the content of individual Properties.
+     *
+     * @param stateStore the source of data to be returned to callers
+     * @param propertyDeserializer a deserializer which can turn any Property in the provided
+     *                             {@code stateStore} to valid JSON
```
Comment (mgummelt) [inline src/main/java/org/apache/mesos/scheduler/api/TaskResource.java]: same format as `StateResource`

File: src/main/java/org/apache/mesos/state/api/StateResource.java (hunk 2 above)
Comment (mgummelt) [inline src/main/java/org/apache/mesos/state/api/StateResource.java]: I removed the reference to properties.  There were some vestiges of copy+pasting this from `StateStore`

**Pull request merged**